### DISCUSSION TITLE
Text Equations to MDF?

### DISCUSSION THREAD

**Discussion by benwandrew** [ModECI/MDF#354] [Q&A]
Hello,

I'm wondering if there is (or will be) functionality for converting text equations representing a model into MDF. Instead of having different export functions for each model type (eg, [PyTorch to MDF](https://mdf.readthedocs.io/en/latest/api/export_format/PyTorch/PyTorch.html#pytorch-to-mdf)), this could encompass any type whose equations can be printed as text.

Thanks!

**Comment by pgleeson** [comment]
Thanks for your interest in MDF @benwandrew. This seems to certainly be in the scope of MDF, I can see an RNN network being trained for such a task, and it could certainly be encoded in MDF. They key thing required would be a training set of written equations/models and their equivalent in some structured, machine readable format. Do you know of such a training set?

**Reply by benwandrew** [reply]
Unfortunately I don't know of such a training set @pgleeson! But I wasn't actually thinking of a ML approach; rather a more explicit / hand-written algorithm. I don't have any concrete ideas for that, however, so I thought I'd ask if your team had considered something along those lines. Thanks for the input!

**Reply by pgleeson** [reply]
@benwandrew We've not done anything in that area and are not really planning to currently. The best I can say is that MDF could potentially be used, but not sure how much better it would be than well written Python classes...

**Reply by benwandrew** [reply]
that makes sense. thank you!
